File: scripts/algorithms/S/Sign of the Product of an Array/Sign of the Product of an Array.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int res = 1;
        for(auto x : nums)
        {
            if(x < 0) res *= -1;
            else if(x == 0) return 0;
        }
        return res;
    }
};

File: scripts/algorithms/S/Sign of the Product of an Array/Sign of the Product of an Array.js
// Runtime: 83 ms (Top 65.37%) | Memory: 43.7 MB (Top 52.05%)

var arraySign = function(nums) {
    // use filter to find total negative numbers in the array
    let negativeCount = nums.filter(n => n<0).length;

    // check if the nums array has zero. If it does, then return 0
    if(nums.includes(0)) return 0;

    // If negativeCount variable is even answer is 1 else -1
    return negativeCount % 2 === 0 ? 1 : -1
};